### PR TITLE
feat: (refactor) introduce uuv-cli engine from runner-commons, #576

### PR DESCRIPTION
**Pull request by luifr10**

### CODE REVIEW

File: packages/runner-playwright/src/lib/runner-playwright.ts [hunk 1]
```diff
@@ -29,100 +28,118 @@
...
-        console.debug(`  -> targetTestFile: ${targetTestFile}`);
+
+    private buildCommand(options: Partial<UUVCliOptions>, configFile: string, reporter: string): string {
+        return [
+            "npx",
+            "playwright",
+            "test",
+            `--project=${options.browser}`,
+            "-c",
+            configFile,
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-playwright/src/lib/runner-playwright.ts:110]: Doesn't came from user input

**Review by stanlee974 — CHANGES_REQUESTED** [review]

File: packages/runner-commons/src/cli/options.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export interface UUVCliOptions {
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-commons/src/cli/options.ts:1]: Corrigé

File: packages/runner-commons/src/cli/helper.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const argv: any = minimist(process.argv.slice(2));
+        if (argv._.length < 1) {
+            throw new Error("No command specified");
+        }
+    }
+
+    static checkTargetTestFiles() {
+        const argv: any = minimist(process.argv.slice(2));
+        const targetTestFile = argv.targetTestFile;
+        if (!isNull(targetTestFile) && !isEmpty(targetTestFile)) {
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-commons/src/cli/helper.ts:38]: Corrigé

File: packages/runner-playwright/src/lib/runner-playwright.ts [hunk 4]
```diff
@@ -29,100 +28,118 @@ export interface UUVPlaywrightCucumberMapItem {
...
+        cp.fork(path.join(__dirname, "watch-test-files"), [this.tempDir, this.projectDir, options.extraArgs.TAGS]);
+        this.runPlaywright(options);
+    }
 
-export async function run(mode: "open" | "e2e", tempDir = "uuv/.features-gen/e2e", configDir = "uuv", argv: any) {
-    const { browser, env, targetTestFile } = extractArgs(argv);
-    await executePreprocessor(tempDir, configDir, env);
-    if (mode === "open") {
-        cp.fork(path.join(__dirname, "watch-test-files"), [tempDir, configDir, env]);
+    private getTargetTestFileForPlawright(targetTestFile?: string): string {
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-playwright/src/lib/runner-playwright.ts:73]: Corrigé

File: packages/runner-commons/src/cli/helper.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        console.debug(`  -> browser: ${options.browser}`);
+        console.debug(`  -> report: ${JSON.stringify(options.report)}`);
+        console.debug(`  -> env: ${JSON.stringify(options.extraArgs)}`);
+        if (options.targetTestFile) {
+            console.debug(`  -> targetTestFile: ${options.targetTestFile}`);
+        }
+        console.debug("\n");
+    }
+
+    static deleteAllFileOfDirectory(dirPath) {
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-commons/src/cli/helper.ts:97]: Corrigé

File: packages/runner-commons/src/cli/engine.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { UUVCliRunner } from "./runner";
+import { UUVCliHelper } from "./helper";
+import chalk from "chalk";
+
+export class UUVCliEngine {
```
[the remark this answers is not in the document]
Comment (luifr10) [reply packages/runner-commons/src/cli/engine.ts:5]: Conservé non static pour le moment

**Pull request merged**